singularity/init.sh

```diff
@@ -3,27 +3,34 @@
 # strict mode so script quits immediately upon failure
 set -eo pipefail
 
-dataDir=$1 ## Directory that contains GENERAL and PROTECTED folders
-containerDataDir=$2 ## Directory to save data generated by dpdash
-serviceHost=$3 ##Optional; all long host names (FQDNs) of rabbitmq and mongodb services.
-
-[ -z $DPDASH_IMG ] && DPDASH_IMG=dpdash.sif
-if [ ! -f $DPDASH_IMG ] && [ ! -d $DPDASH_IMG ]
-then
-	echo $DPDASH_IMG cannot be found. Make sure it exists or define DPDASH_IMG properly.
-	exit 1
+# Getting environment variables from .env
+source ./loadenv.sh
+# Check if required variables are defined
+source ./varcheck.sh
+
+dataDir=$DPDASH_DATA_DIR
+containerDataDir=$DPDASH_STATE_DIR
+
+# use optional vars, or defaults
+dpdashPort=${DPDASH_PORT:-8000}
+mongoPort=${DPDASH_MONGO_PORT:-27017}
+rabbitDist=${DPDASH_RABBIT_DIST:-25671}
+rabbitName=${DPDASH_RABBIT_NAME:-rabbit}
+rabbitPort=${DPDASH_RABBIT_PORT:-5671}
+supervisorId=${DPDASH_SUPERVISOR_ID:-supervisor}
+
+if [ "$(ls -A $containerDataDir)" ]; then
+    echo "$containerDataDir needs to both exist and be empty"
+    exit 1
 fi
 
-if [ -z "$serviceHost" ]
-then
+if [ ! -z ${DPDASH_SERVICE_HOST+x} ]
+then 
+    serviceHost=$DPDASH_SERVICE_HOST
+else
     serviceHost=`hostname -f | xargs`
 fi
 
-if [ -d "$containerDataDir" ] && [ "$(ls -A $containerDataDir)" ]; then
```

Comment:
It was replaced with just the condition `[ "$(ls -A $containerDataDir)" ]`:

https://github.com/PREDICT-DPACC/dpdash/blob/378c872a8d866292f63129564a770a3a070a48f3/singularity/init.sh#L22

since the condition `[ -d "$containerDataDir" ]` already exists in `varcheck.sh`:
https://github.com/PREDICT-DPACC/dpdash/blob/378c872a8d866292f63129564a770a3a070a48f3/singularity/varcheck.sh#L8